pilot/pkg/xds/ep_filters.go

```diff
@@ -106,16 +106,19 @@ func (b *EndpointBuilder) EndpointsByNetworkFilter(endpoints []*LocLbEndpointsAn
 		}
 
 		// Sort the gateways into an ordered list so that the generated endpoints are deterministic.
-		gateways := make([]*model.NetworkGateway, 0, len(gatewayWeights))
+		gateways := make([]model.NetworkGateway, 0, len(gatewayWeights))
 		for gw := range gatewayWeights {
-			gw := gw
-			gateways = append(gateways, &gw)
+			gateways = append(gateways, gw)
 		}
 		gateways = model.SortGateways(gateways)
 
 		// Create endpoints for the gateways.
 		for _, gw := range gateways {
-			epWeight := gatewayWeights[*gw]
+			epWeight := gatewayWeights[gw]
+			if epWeight == 0 {
+				log.Warnf("gateway weight must not be 0, scaleFactor is %d", scaleFactor)
```

Comment:
Should this be `continue`? This seems to occur if our total `weight` to distribute across the gateways isn't large enough to actually utilize all the gateways. 

I think this should be `continue` for safety purposes, with a warning log, but the underlying issue is that scaleFactor isn't making the total weight large enough to utilize all the gws. I think we can test this by making gws > endpoints by some degree?

Reply:
This is just a debug info, in theory it could not enter, because the scaleFactor is the lcm of all gateways